handlers/digital-voucher-api/src/main/scala/com/gu/digital_voucher_api/imovo/ImovoClient.scala

```diff
@@ -49,7 +50,9 @@ object ImovoClient extends LazyLogging {
       val request = body.fold(requestWithoutBody)(b => requestWithoutBody.body(b))
 
       for {
-        response <- EitherT.right[ImovoClientException](request.send())
+        response <- EitherT(request.send().map(_.asRight[ImovoClientException]).recover {
+          case e => Left(ImovoClientException(e.toString))
+        })
```

Comment:
I found something even better:
```suggestion
        response <- request.send().attemptT.leftMap(e => ImovoClientException(e.toString))
```

Reply:
Nice solution! Thanks.  I don't think I was likely to come up with this in a hurry.